### DISCUSSION TITLE
Multi-tenant Application for Telehealth

### DISCUSSION THREAD

**Discussion by Qodestackr** [stack-auth/stack-auth#372] [General]
I have decided to utilize StackAuth in a Practice Management/Patient Engagement platform. Its very intuitive for managing users. However, I also want to show devices that have logged in, for instance a pharma or doctors for granular management of stuff. How can I get it right?

**Comment by fomalhautb** [comment]
Can you elaborate a bit more on where you want to show the devices and how you wan to use them?

**Comment by Qodestackr** [comment]
I want to use custom onboarding flows, where a provider can choose a type: DOCTOR(Indie PRACTICE), NURSE(To support Remote Care), Lab, Pharmacy. But often, Account creator is likely the owner and those working on pharmacies esp those distributed around Nairobi are just pharmacist selling drugs. Basically helping them know of new device logins. Let me know if its possible I stop caring about it and just focus on migrating the auth.

ps: Am I supposed to append roles in clientReadOnlyMetadata? So that it is only editable server side?

**Reply by fomalhautb** [reply]
Got it, we currently don't support that yet. 

Yes, `clientReadOnlyMetadata` is only editable on the server side. You might also want to check if the [permission system]( https://docs.stack-auth.com/concepts/permissions) serves your purpose, so you don't need to implement the roles yourself.

**Comment by madster456** [comment]
A custom onboarding flow can be achieved like this:
[Custom Onboarding Flow](https://docs.stack-auth.com/docs/concepts/user-onboarding)

If you have questions, please do let me know. 

---

<img src="https://stack-auth.com/_next/static/media/stack-auth-logo.84c18f5e.svg" width="200" />
